src/main/java/seedu/mindmymoney/command/UpdateCommand.java

```diff
@@ -72,42 +84,57 @@ private boolean hasIncomeFlag() {
         return updateInput.contains(FLAG_OF_INCOME);
     }
 
+    private boolean hasCategoryFlag() {
+        return updateInput.contains(FLAG_OF_CATEGORY);
+    }
+
     /**
      * Updates an Expenditure entry in user's expenditure list.
      *
      * @throws MindMyMoneyException when an invalid command is received.
      */
     public void updateExpenditure() throws MindMyMoneyException {
         try {
-            String description;
-            String category = null;
+            String newCategory = null;
             String[] parseUpdateInput = updateInput.split(" ", SPLIT_LIMIT);
+            String indexAsString = parseUpdateInput[INDEX_OF_FIRST_ITEM];
+            int indexToUpdate = Integer.parseInt(indexAsString) + LIST_INDEX_CORRECTION;
+            String newPaymentMethod = null;
+
+            // Parse data from user's input
+            if (hasCategoryFlag()) {
```

Comment:
category flag is compulsory, so is all flags, so update command will always just replace all fields